File: packages/graph/__fixtures__/extractor-project/validators.ts
import { z } from 'zod';

export const UserSchema = z.object({
  email: z.string().email(),
  name: z.string().min(1).max(255),
  age: z.number().int().min(0).max(150),
  password: z.string().min(8).regex(/[A-Z]/).regex(/[0-9]/),
});

export const OrderSchema = z.object({
  amount: z.number().positive().max(1000000),
  currency: z.enum(['USD', 'EUR', 'GBP']),
  items: z.array(z.string()).min(1),
});

export const AddressSchema = z.object({
  street: z.string().min(1),
  city: z.string().min(1),
  zip: z.string().regex(/^\d{5}(-\d{4})?$/),
  country: z.string().length(2),
});

File: packages/graph/__fixtures__/extractor-project/validators.rs
use validator::Validate;

#[derive(Validate)]
pub struct User {
    #[validate(email)]
    pub email: String,

    #[validate(length(min = 1, max = 255))]
    pub name: String,

    #[validate(range(min = 0, max = 150))]
    pub age: u32,

    #[validate(length(min = 8))]
    pub password: String,
}

#[derive(Validate)]
pub struct Order {
    #[validate(range(min = 0.01, max = 1000000.0))]
    pub amount: f64,

    #[validate(length(equal = 3))]
    pub currency: String,

    #[validate(length(min = 1))]
    pub items: Vec<String>,
}

#[derive(Validate)]
pub struct Address {
    #[validate(length(min = 1))]
    pub street: String,

    #[validate(length(min = 1))]
    pub city: String,

    #[validate(regex(path = "ZIP_RE"))]
    pub zip_code: String,

    #[validate(length(equal = 2))]
    pub country: String,
}
